client/components/happychat/button.jsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import page from 'page';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import viewport from 'lib/viewport';
+import { openChat } from 'state/ui/happychat/actions';
+import Button from 'components/button';
+import Gridicon from 'components/gridicon';
+
+const HappychatButton = React.createClass( {
+	onOpenChat: function() {
+		const { onOpenChat } = this.props;
+		if ( viewport.isMobile() ) {
+			// For mobile clients, happychat will always use the page compoent instead of the sidebar
+			page( '/me/chat' );
+			return;
```

Comment:
Personal preference, but especially since it's just a single line in the other logical path, seems like `else` would work equally as well and perhaps more clearly:

```js
if ( viewport.isMobile() ) {
	page( '/me/chat' );
} else {
	onOpenChat();
}
```